src/Controller/Component/ThumbsComponent.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * BEdita, API-first content management framework
+ * Copyright 2021 ChannelWeb Srl, Chialab Srl
+ *
+ * This file is part of BEdita: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * See LICENSE.LGPL or <http://gnu.org/licenses/lgpl-3.0.html> for more details.
+ */
+namespace App\Controller\Component;
+
+use BEdita\SDK\BEditaClientException;
+use BEdita\WebTools\ApiClientProvider;
+use Cake\Controller\Component;
+use Cake\Utility\Hash;
+
+/**
+ * Handles thumbs.
+ */
+class ThumbsComponent extends Component
+{
+    /**
+     * Retrieve thumbnails URL of related objects in `meta.url` if present.
+     *
+     * @param array|null $response Related objects response.
+     * @return void
+     */
+    public function urls(?array &$response): void
+    {
+        if (empty($response) || empty($response['data'])) {
+            return;
+        }
+
+        // extract ids of objects
+        $ids = (array)Hash::extract($response, 'data.{n}[type=/images|videos/].id');
+        if (empty($ids)) {
+            return;
+        }
+
+        $getThumbs = function (array $ids): ?array {
```

Comment:
method seems very long - why not create a `protected function getThumbs(array $ids)` ?